worlds/tww/__init__.py

```diff
@@ -0,0 +1,678 @@
+
+    def _determine_progress_and_nonprogress_locations(self) -> tuple[set[str], set[str]]:
+        """
+        Determine which locations are progress and nonprogress in the world based on the player's options.
+
+        :return: A tuple of two sets, the first containing the names of the progress locations and the second containing
+        the names of the nonprogress locations.
+        """
+
+        def add_flag(option: Toggle, flag: TWWFlag) -> TWWFlag:
+            return flag if option else TWWFlag.ALWAYS
+
+        options = self.options
+
+        enabled_flags = TWWFlag.ALWAYS
+        enabled_flags |= add_flag(options.progression_dungeons, TWWFlag.DUNGEON | TWWFlag.BOSS)
+        enabled_flags |= add_flag(options.progression_tingle_chests, TWWFlag.TNGL_CT)
+        enabled_flags |= add_flag(options.progression_dungeon_secrets, TWWFlag.DG_SCRT)
+        enabled_flags |= add_flag(options.progression_puzzle_secret_caves, TWWFlag.PZL_CVE)
+        enabled_flags |= add_flag(options.progression_combat_secret_caves, TWWFlag.CBT_CVE)
+        enabled_flags |= add_flag(options.progression_savage_labyrinth, TWWFlag.SAVAGE)
+        enabled_flags |= add_flag(options.progression_great_fairies, TWWFlag.GRT_FRY)
+        enabled_flags |= add_flag(options.progression_short_sidequests, TWWFlag.SHRT_SQ)
+        enabled_flags |= add_flag(options.progression_long_sidequests, TWWFlag.LONG_SQ)
+        enabled_flags |= add_flag(options.progression_spoils_trading, TWWFlag.SPOILS)
+        enabled_flags |= add_flag(options.progression_minigames, TWWFlag.MINIGME)
+        enabled_flags |= add_flag(options.progression_battlesquid, TWWFlag.SPLOOSH)
+        enabled_flags |= add_flag(options.progression_free_gifts, TWWFlag.FREE_GF)
+        enabled_flags |= add_flag(options.progression_mail, TWWFlag.MAILBOX)
+        enabled_flags |= add_flag(options.progression_platforms_rafts, TWWFlag.PLTFRMS)
+        enabled_flags |= add_flag(options.progression_submarines, TWWFlag.SUBMRIN)
+        enabled_flags |= add_flag(options.progression_eye_reef_chests, TWWFlag.EYE_RFS)
+        enabled_flags |= add_flag(options.progression_big_octos_gunboats, TWWFlag.BG_OCTO)
+        enabled_flags |= add_flag(options.progression_expensive_purchases, TWWFlag.XPENSVE)
+        enabled_flags |= add_flag(options.progression_island_puzzles, TWWFlag.ISLND_P)
+        enabled_flags |= add_flag(options.progression_misc, TWWFlag.MISCELL)
+
+        progress_locations: set[str] = set()
+        nonprogress_locations: set[str] = set()
+        for location, data in LOCATION_TABLE.items():
+            if data.flags & enabled_flags == data.flags:
+                progress_locations.add(location)
+            else:
+                nonprogress_locations.add(location)
+        assert progress_locations.isdisjoint(nonprogress_locations)
+
+        return progress_locations, nonprogress_locations
+
+    def _get_classification_name(self, classification: IC) -> str:
```

Comment:
Does not use the `self` argument so can be decorated as a staticmethod.
```suggestion
    @staticmethod
    def _get_classification_name(classification: IC) -> str:
```